Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/CsvExportProvider.cs

```diff
@@ -1,31 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
+using System.Reflection;
+using System.Reflection.Emit;
 using VirtoCommerce.ExportModule.Core.Model;
 using VirtoCommerce.ExportModule.Core.Services;
 
 namespace VirtoCommerce.ExportModule.Data.Services
 {
-    public class CsvExportProvider : IExportProvider
+    public sealed class CsvExportProvider : IExportProvider
     {
-        private readonly Stream _stream;
+        private static class DynamicMethodGenerator
+        {
+            public delegate object GetMethodDelegate(object source);
```

Comment:
Maybe we could make interface like: 
```
        public interface IPropertyValueGetter
        {
            object GetPropertyValue(object obj, MemberInfo memberInfo);
        }
```
implement it with ILPropertyValueGetter and incapsulate property value get logic and caching there?

Reply:
Not immediately, I think. Let's look how it works. May be later.